src/axom/sidre/tests/sidre_mfem_datacollection.cpp

```diff
@@ -127,6 +127,84 @@ TEST(sidre_datacollection, dc_reload)
   EXPECT_TRUE(sdc_reader.verifyMeshBlueprint());
 }
 
+TEST(sidre_datacollection, dc_qf_reload)
+{
+  //Set up a small mesh and a couple of grid function on that mesh
+  mfem::Mesh *mesh = new mfem::Mesh(2, 3, mfem::Element::QUADRILATERAL, 0, 2.0, 3.0);
+  mfem::FiniteElementCollection *fec = new mfem::LinearFECollection;
+  mfem::FiniteElementSpace *fespace = new mfem::FiniteElementSpace(mesh, fec);
+
+  const int intOrder = 3;
+  const int qs_vdim = 1;
+  const int qv_vdim = 2;
+
+  mfem::QuadratureSpace *qspace = new mfem::QuadratureSpace(mesh, intOrder);
+  // We want Sidre to allocate the data for us and to own the internal data.
+  // If we don't do the below then the data collection doesn't save off the data
+  // structure.
+  mfem::QuadratureFunction *qs = new mfem::QuadratureFunction(qspace, nullptr, qs_vdim);
+  mfem::QuadratureFunction *qv = new mfem::QuadratureFunction(qspace, nullptr, qv_vdim);
+
+  int Nq = qs->Size();
+
+  // The mesh and field(s) must be owned by Sidre to properly manage data in case of
+  // a simulated restart (save -> load)
+  bool owns_mesh = true;
+  MFEMSidreDataCollection sdc_writer(COLL_NAME, mesh, owns_mesh);
+
+  // sdc_writer owns the quadrature function fields and the underlying data
+  sdc_writer.RegisterQField("qs", qs);
+  sdc_writer.RegisterQField("qv", qv);
+
+  // The data needs to be instantiated before we save it off
+  for (int i = 0; i < Nq; ++i)
+  {
+      (*qs)(i) = double(i);
+      (*qv)(2*i+0) = double(i);
+      (*qv)(2*i+1) = double(Nq - i - 1);
+  }
+
+  sdc_writer.SetPrefixPath("/tmp/dc_qf_test");
+  sdc_writer.SetCycle(5);
+  sdc_writer.SetTime(8.0);
+  sdc_writer.Save();
+
+  MFEMSidreDataCollection sdc_reader(COLL_NAME);
+  sdc_reader.SetPrefixPath("/tmp/dc_qf_test");
```

Comment:
Minor: To avoid potential issues on Windows, the new "convention" with these tests is to use the working directory (default prefix path) + the name of the test:
```suggestion
  MFEMSidreDataCollection sdc_reader(testName());
```